modules/images/dominant-color/load.php

```diff
@@ -126,8 +126,12 @@ function dominant_color_img_tag_add_dominant_color( $filtered_image, $context, $
 		$extra_class  = $image_meta['has_transparency'] ? 'has-transparency' : 'not-transparent';
 	}
 
-	if ( ! empty( $data ) || ! empty( $style ) ) {
-		$filtered_image = str_replace( '<img ', '<img ' . $data . $style, $filtered_image );
+	if ( ! empty( $data ) || ! empty( $style ) || ! str_contains( $data, $filtered_image ) ) {
```

Comment:
Why is the `str_contains` check needed here? That change appears to be unrelated to the fix to not duplicate the `style` attribute, which based on the PR title is the goal here.

Also, it looks incorrect: It is impossible for `$data` to ever contain `$filtered_image`, as `$data` is declared just above here. Do these two parameters have to be switched maybe? Regardless of that though, please also clarify why this is needed in the first place. I don't see what it fixes.

Reply:
Curious about this also.

1. `str_contains( $haystack, $needle )` [Ref](https://www.php.net/manual/en/function.str-contains.php), means that this condition is looking for the filtered image markup inside `data-dominant-color="#123"`, which would be impossible. The arguments would need to be switched as Felix says.
2. Regarding the purpose, is it possible that this `str_contains()` is intended to cover a manually added `data-dominant-color: #123` attribute/value pair?

If not, then I'm not sure what the purpose of this condition is.

If so, does this PR account for a `data-dominant-color` attribute being added manually, but not added as a CSS variable in the `style` attribute? If the PR doesn't account for this, should it?

e.g.
```php
$featured_image = '<img src="hello.jpg" alt="Hello.jpg" width="100" height="100" data-dominant-color="#000" style="display: block;">';
```

Reply:
I migrated this back from the core patch 
the idea here is to stop adding the style and data twice Just in case the the_content filter is called twice

Reply:
@pbearne That isn't strictly necessary since `the_content` filter is only called once. The duplicate execution of `wp_filter_content_tags()` is not because of the `the_content`, it comes from block themes and their full page template parsing (see https://github.com/WordPress/wordpress-develop/pull/3549 for a fix).

With that said, #582 implements the addition of the attributes in a cleaner way, clearly separating the logic around the data attributes and the style attribute.

I would advise you revert all changes around the lines here and instead add a condition to the top of this function:
```php
if ( str_contains( $filtered_image, ' data-dominant-color="' ) || str_contains( $filtered_image, ' data-has-transparency="' ) ) {
	return $filtered_image;
}
```

With this logic we holistically ensure nothing is added to the image tag if this function already added its attributes before.